raft/tracker/progress.go

```diff
@@ -101,70 +107,86 @@ func (pr *Progress) BecomeProbe() {
 	}
 }
 
+// BecomeReplicate transitions into StateReplicate, resetting Next to Match+1.
 func (pr *Progress) BecomeReplicate() {
 	pr.ResetState(StateReplicate)
 	pr.Next = pr.Match + 1
 }
 
+// BecomeSnapshot moves the Progress to StateSnapshot with the specified pending
+// snapshot index.
 func (pr *Progress) BecomeSnapshot(snapshoti uint64) {
 	pr.ResetState(StateSnapshot)
 	pr.PendingSnapshot = snapshoti
 }
 
-// maybeUpdate returns false if the given n index comes from an outdated message.
+// MaybeUpdate returns false if the given n index comes from an outdated message.
 // Otherwise it updates the progress and returns true.
 func (pr *Progress) MaybeUpdate(n uint64) bool {
 	var updated bool
 	if pr.Match < n {
 		pr.Match = n
 		updated = true
-		pr.Resume()
+		pr.RecentlyContacted = false
```

Comment:
I think inlining this assignment obscures what's going on - the contact hasn't become any less recent. What's happening is that we've received an ack to our probe. I think it would be better to indirect through a method, called either Resume as before or something like ProbeReceived().

Reply:
I inlined this because I found `Resume()` opaque, but I like `ProbeReceived()`. Actually `ProbeAcked()` is even better I think, going with that.